src/status/chat.nim

```diff
@@ -134,10 +152,33 @@ proc updateContacts*(self: ChatModel, contacts: seq[Profile]) =
     self.contacts[c.id] = c
   self.events.emit("chatUpdate", ChatUpdateArgs(contacts: contacts))
 
+
+proc deleteSpamGroupChats(self: ChatModel, contacts: seq[Profile]) =
+  var chats = status_chat.loadChats()
+  for chat in chats:
+    if chat.chatType == ChatType.PrivateGroupChat:
+      var isContact = false
+      var joined = false
+      for member in chat.members:
+        if member.id == self.publicKey and member.joined:
+          joined = true
+        if member.admin and member.joined:
+          for contact in contacts:
+            if contact.address == member.id:
+              isContact = true
+      if not isContact and not joined:
+        status_chat.deactivateChat(chat)
```

Comment:
If I'm not mistaken, this code black is a duplicate of the one above. Could you move it in a function we could use in both places?